Sources/Hummingbird/Middleware/MiddlewareModule/MiddlewareFixedTypeBuilder.swift

```diff
@@ -64,4 +36,28 @@ public enum MiddlewareFixedTypeBuilder<Input, Output, Context> {
     ) -> _Middleware2<M0, M1> where M0.Input == M1.Input, M0.Output == M1.Output, M0.Context == M1.Context {
         _Middleware2(m0, m1)
     }
+
+    public static func buildOptional<M0: MiddlewareProtocol>(_ component: M0?) -> _ConditionalMiddleware<M0, _EmptyMiddleware<M0>> {
+        if let component {
+            .first(component)
+        } else {
+            .second(_EmptyMiddleware())
+        }
+    }
+
+    public static func buildEither<M0: MiddlewareProtocol, M1: MiddlewareProtocol>(
+        first content: M0
+    ) -> _ConditionalMiddleware<M0, M1> {
+        .first(content)
```

Comment:
Why do we need the `ConditionalMiddleware` type here? Can't you just return `content`? eg
```swift
public static func buildEither<M0: MiddlewareProtocol>(
    first: M0
) -> M0 {
    first
}

public static func buildEither<M1: MiddlewareProtocol>(
    second: M1
) -> M1 {
    second
}
```

Reply:
🤔 I think in some of my personal result builders this was not possible, but yea in this case it is.